docs/source/tutorial/05-connect-queries-to-your-ui.mdx

```diff
@@ -2,56 +2,57 @@
 title: "5. Connect your queries to your UI"
 ---
 
-In this chapter, you are going to display a list of Launch Sites in a [RecyclerView](https://developer.android.com/guide/topics/ui/layout/recyclerview).
+In this chapter, you are going to display a list of Launch Sites in a [LazyColumn](https://developer.android.com/jetpack/compose/lists#lazy).
 
 ## Configure LaunchListAdapter
 
-Open `LaunchListAdapter` and add a `launches` property:
+In `LaunchList`, declare a list of `LaunchListQuery.Launch`, initialized as empty:
 
-```kotlin title="app/src/main/java/com/example/rocketreserver/LaunchListAdapter.kt"
-class LaunchListAdapter(
-    private val launches: List<LaunchListQuery.Launch>
-) : RecyclerView.Adapter<LaunchListAdapter.ViewHolder>() {
+```kotlin title="app/src/main/java/com/example/rocketreserver/LaunchList.kt"
+@Composable
+fun LaunchList(onLaunchClick: (launchId: String) -> Unit) {
+  var launchList by remember { mutableStateOf(emptyList<LaunchListQuery.Launch>()) }
 ```
 
 `LaunchListQuery.Launch` is a typesafe generated model from your LaunchList.graphql query.
 
-## Bind the data to the UI
+## Make a UI for the items
 
-Use the `launches` property to bind items to your adapter:
+Update the `LaunchItem` composable to pass it a `LaunchListQuery.Launch` and display the id:
 
-```kotlin title="app/src/main/java/com/example/rocketreserver/LaunchListAdapter.kt"
-    override fun getItemCount(): Int {
-        return launches.size
-    }
+```kotlin title="app/src/main/java/com/example/rocketreserver/LaunchList.kt"
+@Composable
+private fun LaunchItem(launch: LaunchListQuery.Launch, onClick: (launchId: String) -> Unit) { // highlight-line
+    ListItem(
+        modifier = Modifier.clickable { onClick(launch.id) }, // highlight-line
+        headlineText = {
+            // Mission name
+            Text(text = "Launch ${launch.id}") // highlight-line
```

Comment:
3 highlight lines is a bit too much for me. I was wondering if it was some rendering but. Might be a personal thing though, feel free to ignore